### PR TITLE
refactored test-cmd core to use os::cmd functions

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@deads2k @liggitt PTAL

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
[test]

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -273,7 +273,7 @@ echo "config files: ok"
 export KUBECONFIG="${HOME}/.kube/non-default-config"
 export CLUSTER_ADMIN_CONTEXT=$(oc config view --flatten -o template --template='{{index . "current-context"}}')
 
-
+exit 1
```
Comment (liggitt) [inline hack/test-cmd.sh]: remove debug

File: hack/test-cmd.sh [hunk 2]
```diff
@@ -207,65 +207,65 @@ export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
...
 # log in and set project to use from now on
-oc login --server=${KUBERNETES_MASTER} --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt" -u test-user -p anything
-oc get projects
-oc project project-foo
-[ "$(oc config view | grep current-context | grep project-foo/${API_HOST}:${API_PORT}/test-user)" ]
-[ "$(oc whoami | grep 'test-user')" ]
-[ "$(oc whoami --config="${MASTER_CONFIG_DIR}/admin.kubeconfig" | grep 'system:admin')" ]
-[ -n "$(oc whoami -t)" ]
-[ -n "$(oc whoami -c)" ]
+os::cmd::expect_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything"
```
Comment (deads2k) [inline hack/test-cmd.sh]: don't hide this output

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -207,65 +207,65 @@ export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
...
+os::cmd::expect_success 'oc get projects'
```
Comment (deads2k) [inline hack/test-cmd.sh]: don't hide this output

File: hack/test-cmd.sh [hunk 4]
```diff
@@ -207,65 +207,65 @@ export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
...
+os::cmd::expect_success 'oc project project-foo'
```
Comment (deads2k) [inline hack/test-cmd.sh]: don't hide this output

**Comment by deads2k** [comment]
I'm fine with this in concept.  I'd like to keep the project context exposed.

**Comment by deads2k** [comment]
re[test]

File: hack/test-cmd.sh [hunk 5]
```diff
@@ -188,18 +189,17 @@ atomic-enterprise start \
   --images="${USE_IMAGES}"
 
 # ensure that DisabledFeatures aren't written to config files
-! grep -i '\<disabledFeatures\>' \
-	"${MASTER_CONFIG_DIR}/master-config.yaml" \
-	"${BASETMPDIR}/atomic.local.config/master/master-config.yaml" \
-	"${NODE_CONFIG_DIR}/node-config.yaml"
+os::cmd::expect_success_and_not_text "cat ${MASTER_CONFIG_DIR}/master-config.yaml" '\<disabledFeatures\>'
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: This failed:
```
FAILURE after 0.010s: hack/test-cmd.sh:192: executing 'cat /tmp/openshift-cmd/openshift.local.config/master/master-config.yaml' expecting success and not text '\<disabledFeatures\>': the output content test failed
Standard output from the command:
apiLevels:
- v1
apiVersion: v1
assetConfig:
  extensionDevelopment: false
  extensionScripts: null
  extensionStylesheets: null
  extensions: null
  loggingPublicURL: ""
  logoutURL: ""
  masterPublicURL: https://127.0.0.1:28443
  metricsPublicURL: ""
  publicURL: https://127.0.0.1:28443/console/
  servingInfo:
    bindAddress: 127.0.0.1:28443
    bindNetwork: tcp4
    certFile: master.server.crt
    clientCA: ""
    keyFile: master.server.key
    maxRequestsInFlight: 0
    namedCertificates: null
    requestTimeoutSeconds: 0
controllerLeaseTTL: 0
controllers: '*'
corsAllowedOrigins:
- 127.0.0.1
- 127.0.0.1:28443
- localhost
disabledFeatures: null
...
There was no error output from the command.
```

Why was this working before? Why were there angle brackets in the `grep` string?

**Comment by stevekuznetsov** [comment]
Updated to expect `disabledFeatures: null` as per https://github.com/openshift/origin/pull/3953#discussion_r50415503

File: hack/test-cmd.sh [hunk 6]
```diff
@@ -191,18 +192,16 @@ atomic-enterprise start \
   --images="${USE_IMAGES}"
 
 # ensure that DisabledFeatures aren't written to config files
-! grep -i '\<disabledFeatures\>' \
-	"${MASTER_CONFIG_DIR}/master-config.yaml" \
-	"${BASETMPDIR}/atomic.local.config/master/master-config.yaml" \
-	"${NODE_CONFIG_DIR}/node-config.yaml"
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: @miminar any reason why `NodeConfig` would contain disabled features? The struct [here](https://github.com/openshift/origin/blob/master/pkg/cmd/server/api/types.go#L66) doesn't have any mention of them. I've removed that check.

**Comment by deads2k** [comment]
```
Running hack/test-cmd.sh:229: executing 'oc login --token=wjyu7G9C9g2fg9KZigYi-Cf_piUfRmG-6yoSX1QAyWE' expecting failure and text 'using the token provided'...
FAILURE after 0.063s: hack/test-cmd.sh:229: executing 'oc login --token=wjyu7G9C9g2fg9KZigYi-Cf_piUfRmG-6yoSX1QAyWE' expecting failure and text 'using the token provided': the command returned the wrong error code
Standard output from the command:
Logged into "https://127.0.0.1:28443" as "test-user" using the token provided.

You don't have any projects. You can try to create a new project, by running

    $ oc new-project <projectname>

There was no error output from the command.
[INFO] Stacked plot for log subset "iops" written to /tmp/openshift-cmd/logs/iops.pdf
```

File: hack/test-cmd.sh (hunk 6 above)
Comment (miminar) [inline hack/test-cmd.sh]: No reason. I'm fine with its removal.

**Comment by stevekuznetsov** [comment]
@deads2k we're green

**Comment by deads2k** [comment]
[merge]

**Pull request merged**